### PR TITLE
feat(LinstorSR): use a specific resource group for DB and HA

### PR DESCRIPTION
**Pull request by Wescoeur**

### CODE REVIEW

File: drivers/linstorvolumemanager.py [hunk 1]
```diff
@@ -1359,6 +1372,7 @@ def destroy(self):
 
             # 4.5. Destroy group and storage pools.
             self._destroy_resource_group(self._linstor, self._group_name)
+            self._destroy_resource_group(self._linstor, self._build_ha_goup_name(self._base_group_name))
```
Comment (Nambrok) [inline drivers/linstorvolumemanager.py]: It is weird having a call here to get the HA group name.

File: drivers/linstorvolumemanager.py [hunk 2]
```diff
@@ -1744,7 +1768,8 @@ def _create_sr(
         driver_pool_name = group_name
         base_group_name = group_name
         group_name = cls._build_group_name(group_name)
-        pools = lin.storage_pool_list_raise(filter_by_stor_pools=[group_name])
+        storage_pool_name = group_name
+        pools = lin.storage_pool_list_raise(filter_by_stor_pools=[storage_pool_name])
         pools = pools.storage_pools
```
Comment (Nambrok) [inline drivers/linstorvolumemanager.py]: Could be done in one line to avoid changing `pools` type.

File: drivers/linstorvolumemanager.py [hunk 3]
```diff
@@ -1825,49 +1850,28 @@ def _create_sr(
...
+              group_name,
+              storage_pool_name,
+              redundancy,
+              True
+            )
+            cls._create_resource_group(
+              lin,
+              ha_group_name,
+              storage_pool_name,
+              3,
```
Comment (Nambrok) [inline drivers/linstorvolumemanager.py:1894]: Hardcoded value?

File: drivers/linstorvolumemanager.py [hunk 4]
```diff
@@ -2149,27 +2154,57 @@ def _get_storage_pools(self, force=False):
...
+        size,
+        place_resources,
+        high_availability
     ):
         size = self.round_up_volume_size(size)
         self._mark_resource_cache_as_dirty()
 
+        group_name = self._group_name
+        if high_availability:
+            group_name = self._build_ha_goup_name(self._base_group_name)
```
Comment (Nambrok) [inline drivers/linstorvolumemanager.py]: `self._build_ha_group_name()`

File: drivers/linstorvolumemanager.py [hunk 5]
```diff
@@ -2849,6 +2943,12 @@ def _build_group_name(cls, base_name):
         # `VG/LV`. "/" is not accepted by LINSTOR.
         return '{}{}'.format(cls.PREFIX_SR, base_name.replace('/', '_'))
 
+    # Used to store important data in a HA context,
+    # i.e. a replication count of 3.
+    @classmethod
+    def _build_ha_goup_name(cls, base_name):
```
Comment (Nambrok) [inline drivers/linstorvolumemanager.py]: `group`

File: drivers/linstorvolumemanager.py (hunk 3 above)
Comment (Wescoeur) [reply drivers/linstorvolumemanager.py:1894]: Yes. Fixed for each installation. No reason to use another value for the replication.

**Pull request merged**